src/Microsoft.AspNet.Http.Abstractions/HttpResponse.cs

```diff
@@ -24,9 +25,9 @@ public abstract class HttpResponse
 
         public abstract bool HasStarted { get; }
 
-        public abstract void OnResponseStarting(Action<object> callback, object state);
+        public abstract void OnResponseStarting(Func<object, Task> callback, object state);
```

Comment:
I filed https://github.com/aspnet/HttpAbstractions/issues/333 to track.

Lou said no sync sugar anywhere as part of agreeing to full async.  I guess we are strongly encouraging async this way. I'll add a stateless sugar overload now